### PR TITLE
Feature/hidden cards

### LINKED ISSUE #2: add deck unit tests
add more deck unit tests

### LINKED ISSUE #7: implement last card is hidden

### PR DESCRIPTION
**Pull request by omerholtzman**

### CODE REVIEW

File: players/pocker_ai.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 
 class PokerAi(ABC):
+    def __init__(self, player_number: bool):
+        self.player_number: bool = player_number
```
Comment (galord123) [inline players/pocker_ai.py]: why player number is not an int?
if it is a bool, maybe `is_first_player`

File: tests/test_hand.py [hunk 2]
```diff
@@ -1,169 +1,84 @@
...
-            Card(Suit.HEARTS, 12),
-        ])
-
-        self.assertFalse(hand > other_hand)
-        self.assertTrue(other_hand > hand)
+from game import Hand, Card
+from game.card import Suit
+
+
+class TestHand(unittest.TestCase):
```
Comment (galord123) [inline tests/test_hand.py:7]: in all cases you initialize `Hand` to be `True`.
what about the case where it is `False`? where do you check it?

File: tests/test_hand_logic.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    def test_flush(self):
+        low_hearts_flush = [Card(Suit.HEARTS, i) for i in range(2, 11, 2)]
+        higher_clubs_flush = [Card(Suit.CLUBS, i) for i in range(3, 12, 2)]
+        higher_diamonds_flush = [Card(Suit.DIAMONDS, i) for i in range(3, 12, 2)]
+
+        self.assertTrue(has_better_cards(higher_diamonds_flush, higher_clubs_flush))
+        self.assertTrue(has_better_cards(higher_diamonds_flush, low_hearts_flush))
+        self.assertTrue(has_better_cards(higher_clubs_flush, low_hearts_flush))
+
+    def test_strait_flush(self):
```
Comment (galord123) [inline tests/test_hand_logic.py]: misspelling of `straight`

File: game/deck.py [hunk 4]
```diff
@@ -1,34 +1,81 @@
...
+            self._cards_left: List[Card] = [Card(suit, number) for suit in SUITS for number in range(1, 14)]
+            self.shuffle()
 
-        self.cards_left_amount = len(self.cards_left)
-        self.suits_left: Dict[Suit, int] = {suit: 13 for suit in SUITS}
-        self.number_left: Dict[int, int] = {number: 4 for number in range(1, 14)}
+            self.cards_left_amount = len(self._cards_left)
+            self._suits_left: Dict[Suit, int] = {suit: 13 for suit in SUITS}
+            self._number_left: Dict[int, int] = {number: 4 for number in range(1, 14)}
+            self._decks: List["Deck"] = []
```
Comment (galord123) [inline game/deck.py:20]: just out of curiosity, if you are having each deck to hold 2 other decks, when their cards are below 12, they want create 2 extra deck themselves?
or am I missing something?

File: players/pocker_ai.py (hunk 1 above)
Comment (omerholtzman) [reply players/pocker_ai.py]: I think a bool will do

File: tests/test_hand.py (hunk 2 above)
Comment (omerholtzman) [reply tests/test_hand.py:7]: solved

File: game/deck.py (hunk 4 above)
Comment (omerholtzman) [reply game/deck.py:20]: yes, what you miss is that the deck creation occurs only using pop and only when we have 12 cards there at the pop-stage. any way - those cards will never have 13 cards neither will getting popped

**Pull request merged**